logging_policy.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+import numpy as np
+
+### Simpler Logging Policy
+''' Approach:
+- Only consider the DecisionTeamId - basically, the logging policy assumes that each team has a static Pick distribution that doesn't account for the rest of the context
+- Zero out unavailable maps, reweight toward others
+- Set a minimum probability of 1/num_samples for available maps so that pi/pi_0 is never zero.'''
+
+class LoggingPolicy(object):
+    ''' Logging Policy constructs p(a|x) where x is the DecisionTeamId.
+    Init Inputs:
+	- full_context: DataFrame containing the context features. Must contain the is_available columns for all 7 maps and DecisionTeamId. Other features are not used.
+	- full_action: Series of ints corresponding to chosen map. Order should align with full_context 
+
+	Use:
+	- get_pa_x(context) returns p(a|x) accounting for unavailable maps. Should be used for Importance Weighting.
+    '''
+    def __init__(self,full_context,full_action):
+        self.full_context = full_context
+        self.full_action = full_action
+        self.pa_x_dict = self.calculate_p_action_conditional()
+        self.map_cols = ['de_dust2_is_available', 'de_inferno_is_available',
+                         'de_mirage_is_available', 'de_nuke_is_available',
+                         'de_overpass_is_available', 'de_train_is_available',
+                         'de_vertigo_is_available']
+    
+    def calculate_p_action_conditional(self):
+        # Calculate p(a|x) where x is the DecisionTeamId.
+        # Sets minimum p(a|x) = 1/num_samples
+        pa_x_dict = {}
+        for team in self.full_context['DecisionTeamId'].unique():
+            a_x = []
+            for action in self.full_action.unique():
+                a_x.append(max(1,(self.full_action[self.full_context['DecisionTeamId']==team]==action).sum()))
+                pa_x = [ax/np.sum(a_x) for ax in a_x]
```

Comment:
i think it might be clearer to filter the context/action df's to only the currently-looked-at team, then use `df.value_counts(normalize=True)` (or even without normalizing and then just dividing after adding 1, like what you did here)

Reply:
Ahh, the reason I did not is that when the count = 0 the action doesn't show up.  So this way ensured that all actions were represented. There's a better way to do it I'm sure, but (to your point) it only takes a few seconds to calculate anyways.